src/components/App.js

```diff
@@ -5,18 +5,48 @@
 import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
-import { calculate } from '../logic/calculate';
+import calculate from '../logic/calculate';
 import '../styles/App.css';
 
-const App = props => (
-  <div>
-    <Display calcResult={props.result} />
-    <ButtonPanel />
-  </div>
-);
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
 
-App.defaultProps = {
-  result: '0',
-};
+    this.state = {
+      display: null,
+      total: null,
+      next: null,
+      operation: null,
+    };
 
-export default App;
\ No newline at end of file
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(btnName) {
+    const calcData = {
+      display: this.state.display,
+      total: this.state.total,
+      next: this.state.next,
+      operation: this.state.operation,
+    };
+
+    const calcValue = calculate(calcData, btnName);
+
+    this.setState({
+      display: calcValue.display,
+      total: calcValue.total,
+      next: calcValue.next,
+      operation: calcValue.operation,
+    });
+  }
+
+  render() {
+    const { total, next, operation } = this.state;
+    return (
+      <div>
+        <Display value={this.state.display || this.props.value} />
+        <ButtonPanel clickHandler={this.handleClick} />
```

Comment:
(`line 47`) : According to the requirement, you had to pass either `total` or `next` value from state to `<Display />` component [[see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passToatalToDisplay.png)]

Reply:
@shloch,

Thanks for your feedback. I have fixed the bug above.

About the `display` property on `state` and using 
```<Display value={this.state.display || this.props.value} />
```
makes more sense to me, as I struggled to reason about how to use `total` for showing the operation keys. I think it is a better design to think of `display` as a separate property in `state`, instead of being hacky about using `total`. I checked my partner's code that used `total` in `<Display />` and I found that the logic is split into three places, inside `handleClick` function, inside `render` function as well inside `calculate` function. Using `display` helped me separate the logic to only one place - inside `calculate` function alone

I couldn't find you on Slack, it shows you have been deactivated there. I would have liked to point out why using `total` doesn't make sense to me